### PR TITLE
add trigger keywords in a configuration

### PR DESCRIPTION
**Pull request by ruan11223344**
Sometimes, too many useless messages can also trigger API abuse, so it is important to add a keyword to control whether to reply. Note that the keyword should appear in the first position.

Next, I am thinking about how to automatically update the token problem, the token will expire in about 2 hours.

### CODE REVIEW

File: src/chatgpt.ts [hunk 1]
```diff
@@ -41,7 +42,11 @@ export class ChatGPTBot {
             email?: string;
             password?: string;
             session_token?: string;
+            trigger_keywords?:string;
           }): Promise<string> => {
+            if(account.trigger_keywords){
```
[the remark this answers is not in the document]
Comment (ruan11223344) [reply src/chatgpt.ts:47]: If trigger keywords are not set, any message will be replied to by default

File: src/chatgpt.ts [hunk 2]
```diff
@@ -152,7 +169,7 @@ export class ChatGPTBot {
     );
     console.log(`Hit GPT Enabled Group: ${topic} in room: ${room.id}`);
     const response = await this.getGPTMessage(realText, talker.id);
-    const result = `${realText}\n ------\n ${response}`;
+    const result = `${realText}\n ------\n@${talker.name()} ${response}`;
```
[the remark this answers is not in the document]
Comment (ruan11223344) [reply src/chatgpt.ts:172]: all right we can remove it

File: src/chatgpt.ts [hunk 3]
```diff
@@ -4,12 +4,17 @@ import { config } from "./config.js";
...
 import { Cache } from "./cache.js";
 import { ContactInterface, RoomInterface } from "wechaty/impls";
+import {string} from "yaml/dist/schema/common/string";
 
 const SINGLE_MESSAGE_MAX_SIZE = 500;
 export class ChatGPTBot {
   // Record talkid with conversation id
   conversations = new Map<string, ChatGPTConversation>();
   chatGPTPools: Array<ChatGPTAPI> | [] = [];
+  botConfig: { [key: string]: string } = {
```
Comment (fuergaosi233) [inline src/chatgpt.ts:14]: Maybe you could use `Record<string,string> `?

File: src/chatgpt.ts [hunk 4]
```diff
@@ -64,6 +69,9 @@ export class ChatGPTBot {
       });
     console.log(`Chatgpt pool size: ${chatGPTPools.length}`);
     this.chatGPTPools = chatGPTPools;
+    config.botConfig.forEach((item: { [key: string]: string; }) => {
```
Comment (fuergaosi233) [inline src/chatgpt.ts:72]: `this.botConfig = {...config.botConfig,...this.botConfig}`?

**Comment by fuergaosi233** [comment]
Thanks 🙏 But there are a few glitches that can be fixed.

File: src/chatgpt.ts (hunk 4 above)
Comment (ruan11223344) [reply src/chatgpt.ts:72]: modify to 
`
`

Comment (ruan11223344) [reply src/chatgpt.ts:72]: `
   config.botConfig.forEach((item: { [key: string]: string; }) => {
        Object.keys(item).forEach((configItemKey: string ) =>{
            this.botConfig[configItemKey] = item[configItemKey]
        })
    });
`

File: src/chatgpt.ts (hunk 3 above)
Comment (ruan11223344) [reply src/chatgpt.ts:14]: botConfig: Record<string,string> = {
      trigger_keywords:""
  };

File: src/chatgpt.ts (hunk 4 above)
Comment (fuergaosi233) [reply src/chatgpt.ts:72]: Try use `Object.assign({}, config.botConfig, this.botConfig)` pls.

Comment (ruan11223344) [reply src/chatgpt.ts:72]: The problem has been fixed.

**Comment by fuergaosi233** [comment]
LGTM Thanks 🙏

**Pull request merged**